src/main/client/admin.c

```diff
@@ -31,124 +31,165 @@
+    if (!user) {
+        as_error_update(&err, AEROSPIKE_ERR_PARAM, STR_CONVERSION_ERROR_MSG,
+                        "Username");
+        goto CLEANUP_AND_RAISE_EXCEPTION;
+    }
+
+    const char *password = NULL;
+    if (py_password) {
+        password = convert_pyobject_to_str(py_password);
+        if (!password) {
+            as_error_update(&err, AEROSPIKE_ERR_PARAM, STR_CONVERSION_ERROR_MSG,
+                            "Password");
+            goto CLEANUP_AND_RAISE_EXCEPTION;
+        }
     }
 
-    password = PyUnicode_AsUTF8(py_password);
-
     // Convert python object to policy_admin
+    as_policy_admin admin_policy;
+    as_policy_admin *admin_policy_p = NULL;
     pyobject_to_policy_admin(self, &err, py_policy, &admin_policy,
                              &admin_policy_p, &self->as->config.policies.admin);
     if (err.code != AEROSPIKE_OK) {
-        goto CLEANUP;
+        goto CLEANUP_AND_RAISE_EXCEPTION;
     }
 
     // Invoke operation
     Py_BEGIN_ALLOW_THREADS
-    aerospike_create_user(self->as, &err, admin_policy_p, user, password,
-                          (const char **)roles, roles_size);
+    if (password) {
+        aerospike_create_user(self->as, &err, admin_policy_p, user, password,
+                              (const char **)roles, roles_size);
+    }
+    else {
+        aerospike_create_pki_user(self->as, &err, admin_policy_p, user,
+                                  (const char **)roles, roles_size);
+    }
     Py_END_ALLOW_THREADS
 
-CLEANUP:
+CLEANUP_AND_RAISE_EXCEPTION:
     for (int i = 0; i < roles_size; i++) {
         if (roles[i])
             cf_free(roles[i]);
     }
 
+RAISE_EXCEPTION:
     if (err.code != AEROSPIKE_OK) {
         raise_exception(&err);
         return NULL;
     }
+    if (PyErr_Occurred()) {
+        return NULL;
+    }
 
     return PyLong_FromLong(0);
 }
 
+/**
+ *******************************************************************************************************
+ * Create a user in the Aerospike DB.
+ *
+ * @param self                  AerospikeClient object
+ * @param args                  The args is a tuple object containing an argument
+ *                              list passed from Python to a C function
+ * @param kwds                  Dictionary of keywords
+ *
+ * Returns an integer status. 0(Zero) is success value.
+ * In case of error,appropriate exceptions will be raised.
+ *******************************************************************************************************
+ */
+PyObject *AerospikeClient_Admin_Create_User(AerospikeClient *self,
+                                            PyObject *args, PyObject *kwds)
+{
+    // Python Function Arguments
+    PyObject *py_policy = NULL;
+    PyObject *py_user = NULL;
+    PyObject *py_password = NULL;
+    PyObject *py_roles = NULL;
+
+    // Python Function Keyword Arguments
+    static char *kwlist[] = {"user", "password", "roles", "policy", NULL};
```

Comment:
Here the keyword for the user is `user` but in the client.rst docs it is `username` should the docs change?